pkg/aws/ec2/ec2.go

```diff
@@ -347,12 +361,18 @@ func (c *Client) GetSubnets(ctx context.Context) (ipamTypes.SubnetMap, error) {
 }
 
 // CreateNetworkInterface creates an ENI with the given parameters
-func (c *Client) CreateNetworkInterface(ctx context.Context, toAllocate int32, subnetID, desc string, groups []string) (string, *eniTypes.ENI, error) {
+func (c *Client) CreateNetworkInterface(ctx context.Context, toAllocate int32, subnetID, desc string, groups []string, allocatePrefixes bool) (string, *eniTypes.ENI, error) {
+
 	input := &ec2.CreateNetworkInterfaceInput{
-		Description:                    aws.String(desc),
-		SecondaryPrivateIpAddressCount: aws.Int32(toAllocate),
-		SubnetId:                       aws.String(subnetID),
-		Groups:                         groups,
+		Description: aws.String(desc),
+		SubnetId:    aws.String(subnetID),
+		Groups:      groups,
+	}
+	if allocatePrefixes {
+		input.Ipv4PrefixCount = aws.Int32(int32(ipPkg.PrefixCeil(int(toAllocate), option.ENIPDBlockSizeIPv4)))
+		log.Infof("Creating interface with %v prefixes", input.Ipv4PrefixCount)
```

Comment:
do we really need this log in client library here ?

Reply:
Converted this log line to debug level.